### PR TITLE
Create history service backend

### LINKED ISSUE #83: Create history service (backend)
- New table and backend to store userid -> [questionid1, questionid2] etc.

### PR DESCRIPTION
**Pull request by tryyang2001**
# Pull Request

## Description

Create the backend server for history. Right now it supports 3 endpoints:
- `POST /history`
- `GET /history?userId=...&questionId=...`
- `GET /history/user/:userId/question/:questionId/code`
- `PUT /history/user/:userId/question/:questionId/code`
- `DELETE /history/user/:userId/question/:questionId`

We might not need to use the delete endpoint, I created this endpoint so that we can let the user to delete any history if he wanted to do so. 

## Related Issue(s)

Closes #83 

## Changes Made

- create new schema for history, and that schema references other schemas (user & question)
- create API for history service
- create unit tests

## Checklist
- [x] I have checked that the changes included in the PR are intended to merge to `master` or any destination branch.
- [x] I have verified that the new changes do not break any existing functionalities, unless the new changes are intended and have approved by the team.
- [x] I will take care of the merging and delete the side-branch after the PR is merged.

## Additional Notes/References

History service prisma schema must contain User & Question so that it won't overwrite the existing database. I think this is the drawback of having same database for multiple microservices. This also indicates that any future update to User and Question database schema needs to update History schema as well.

For details on how to set up the history service locally, please go through the readme.

### CODE REVIEW

File: backend/history-service/src/middleware/cors.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import cors from "cors";
+
+// TODO: Add production site to allowed origins
+const allowedOrigins = ["http://localhost:3000"];
```
Comment (c0j0s) [inline backend/history-service/src/middleware/cors.ts:4]: Just to let you know, after we deploy to the cloud, we will need to add amplify URL to origin list, so I change to this and let github actions to handle it instead.
```
const allowedOrigins =
  process.env.CORS_ALLOWED_ORIGINS
    ? process.env.CORS_ALLOWED_ORIGINS.split(",")
    : ["http://localhost:3000"];
```

**Review by c0j0s — APPROVED** [review]
LGTM

File: backend/history-service/src/middleware/cors.ts (hunk 1 above)
Comment (tryyang2001) [reply backend/history-service/src/middleware/cors.ts:4]: Sure, I will update this!

**Comment by tryyang2001** [comment]
Thanks for reviewing. Unfortunately, I overlooked one important thing. We have multiple languages, and I believe each user can make multiple submissions for different languages for the same question, so I need to update my code to accommodate this...

**Comment by tryyang2001** [comment]
I am going to merge this PR by tonight if you have no more feedback, thanks!

**Pull request merged**